src/interaction/include/exaDEM/interaction/classifier_analyses.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <iomanip>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <mpi.h>
+#include <exaDEM/interaction/classifier.hpp>
+
+
+namespace exaDEM
+{
+  using namespace exanb;
+
+  template<typename GridT>
+    inline bool filter_duplicates(GridT& G, exaDEM::Interaction& I)
+    {
+      if(I.type < 4) // polyhedron - polyhedron or sphere - sphere
+      {
+        if(G.is_ghost_cell(I.cell_j) && I.id_i > I.id_j)
+        {
+          return false;
+        }
+      }
+      return true;
+    }
+
+  /** CPU only */
+  template<typename GridT>
+    std::stringstream create_buffer(GridT& grid, Classifier& ic)
+    {
+      std::stringstream stream;
+      const int ntypes = ic.number_of_waves();
+      for( int i = 0 ; i < ntypes ; i++ )  
+      {
+        auto [i_ptr, size] = ic.get_info(i);
+        auto [co_ptr, fn_ptr, ft_ptr] = ic.buffer_p(i); 
+
+        for(size_t idx = 0 ; idx < size ; idx++)
+        {
+          Vec3d fn = fn_ptr[idx];
+          Vec3d ft = ft_ptr[idx];
+          /** filter empty interactions */
+          if(fn != Vec3d{0,0,0} || ft != Vec3d{0,0,0})
```

Comment:
now you can just use the d_n variable to test if the interaction is active or not

Reply:
done